istioctl/cmd/tag.go

```diff
@@ -0,0 +1,400 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/spf13/cobra"
+	admit_v1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"istio.io/api/label"
+)
+
+const (
+	// TODO(monkeyanator) move into istio/api
+	istioTagLabel             = "istio.io/tag"
+	istioInjectionWebhookName = "sidecar-injector.istio.io"
+	revisionTagNamePrefix     = "istio-revision-tag"
+)
+
+var (
+	// revision to point tag webhook at
+	revision         = ""
+	overwrite        = false
+	skipConfirmation = false
+)
+
+func tagCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "tag",
+		Short: "Command group used to interact with revision tags",
```

Comment:
Needs more short and long help text to help users understand that tags themselves are **injection** tags, and used to control which namespaces connect to which control plane in a multi-control plane or upgrade scenario.